saleor/graphql/account/filters.py

```diff
@@ -21,34 +21,35 @@ def filter_placed_orders(qs, _, value):
     return filter_range_field(qs, "orders__created__date", value)
 
 
-def filter_status(qs, _, value):
+def filter_staff_status(qs, _, value):
     if value == StaffMemberStatus.ACTIVE:
-        qs = qs.filter(is_staff=True, is_active=True)
-    elif value == StaffMemberStatus.DEACTIVATED:
-        qs = qs.filter(is_staff=True, is_active=False)
+        return qs.filter(is_staff=True, is_active=True)
+    if value == StaffMemberStatus.DEACTIVATED:
+        return qs.filter(is_staff=True, is_active=False)
     return qs
 
 
-def filter_staff_search(qs, _, value):
-    search_fields = (
-        "email",
-        "first_name",
-        "last_name",
-        "default_shipping_address__first_name",
-        "default_shipping_address__last_name",
-        "default_shipping_address__city",
-        "default_shipping_address__country",
-        "default_shipping_address__phone",
-    )
+def filter_user_search(qs, _, value):
     if value:
-        qs = filter_by_query_param(qs, value, search_fields)
+        address = Address.objects.filter(
+            Q(first_name__icontains=value)
+            | Q(last_name__icontains=value)
+            | Q(city__icontains=value)
+            | Q(country__icontains=value)
+            | Q(phone=value)
+        ).values("id")
+        qs = qs.filter(
+            Q(email__icontains=value)
+            | Q(first_name__icontains=value)
+            | Q(last_name__icontains=value)
+            | Q(Exists(address.filter(default_user_shipping=OuterRef("id"))))
```

Comment:
If we implement search by addresses instead of searching by `default_shipping_address` we could also drop `saleor/account/migrations/0051_alter_user_default_shipping_address.py` this migration and changes related to this migration.